test/e2e/hsts_policy_test.go

```diff
@@ -0,0 +1,183 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Helper function for int32 pointers
+func intPtr(s int32) *int32 {
+	return &s
+}
+
+func TestHstsPolicyWorks(t *testing.T) {
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "hsts-policy"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+
+	// Setup a Required HSTS Policy for the ingress config
+	maxAgePolicy := configv1.MaxAgePolicy{LargestMaxAge: intPtr(99999), SmallestMaxAge: intPtr(0)}
+	domainPatterns := []string{domain}
+	hstsPolicy := configv1.RequiredHSTSPolicy{
+		DomainPatterns:          domainPatterns, // this policy will only validate routes with hosts in the DomainPatterns
+		PreloadPolicy:           configv1.RequirePreloadPolicy,
+		IncludeSubDomainsPolicy: configv1.RequireIncludeSubDomains,
+		MaxAge:                  maxAgePolicy,
+	}
```

Comment:
```suggestion
	hstsPolicy := configv1.RequiredHSTSPolicy{
		DomainPatterns:          []string{"*." + dnsConfig.Spec.BaseDomain}, // this policy will only validate routes with hosts in the DomainPatterns
		PreloadPolicy:           configv1.RequirePreloadPolicy,
		IncludeSubDomainsPolicy: configv1.RequireIncludeSubDomains,
		MaxAge: configv1.MaxAgePolicy{
			LargestMaxAge:  intPtr(99999),
			SmallestMaxAge: intPtr(0),
		},
	}
```

We don't create any routes that would use `icName` so this seems redundant.

Reply:
done